File: javascript/949. Largest Time for Given Digits.js
/**
 * @param {number[]} A
 * @return {string}
 */
var largestTimeFromDigits = function(A) {
    const digit = 4;
    let time = -1;
    for(let i=0; i<digit; i++){
        for(let j=0; j<digit; j++){
            if(j!=i){
                for(let k=0; k<digit; k++){
                    if(k!=j && i!=k){
                        let l = 6-i-j-k;
                        let hour = 10*A[i] + A[j];
                        let minute = 10*A[k] + A[l];
                        if(hour < 24 && minute < 60){
                            time = Math.max(time, hour * 60 + minute);
                        }
                    }
                }
            }
        }
    }
    let hours = (Math.floor(time/60) < 10) ? "0" +Math.floor(time/60) : Math.floor(time/60); 
    let minutes = (time%60 < 10) ? "0"+ time%60 : time%60; 
    return time >=0 ? hours + ":" + minutes : "";
};

File: java/949. Largest Time for Given Digits.java
class Solution {
 int time;
 private static final int DIGIT = 4;

 public String largestTimeFromDigits(int[] A) {
  time = -1;

  for (int i = 0; i < DIGIT; i++) {
   for (int j = 0; j < DIGIT; j++) {
    if (i != j) {
     for (int k = 0; k < DIGIT; k++) {
      if (j != k && i != k) {
       int l = 6 - i - j - k;
       int hour = A[i] * 10 + A[j];
       int minute = A[k] * 10 + A[l];
       if (hour < 24 && minute < 60) {
        time = Math.max(time, hour * 60 + minute);
       }
      }
     }
    }

   }

  }
  return time >= 0 ? String.format("%02d:%02d", time / 60, time % 60) : "";
 }
}
